src/libraries/System.Text.RegularExpressions/gen/UpgradeToRegexGeneratorCodeFixer.cs

```diff
@@ -244,58 +218,38 @@ static IEnumerable<ISymbol> GetAllMembers(ITypeSymbol? symbol)
                 }
             }
 
-            // Helper method that searches the passed in property bag for the property with the passed in name, and if found, it converts the
-            // value to an int.
-            static int? TryParseInt32(ImmutableDictionary<string, string?> properties, string name)
-            {
-                if (!properties.TryGetValue(name, out string? value))
-                {
-                    return null;
-                }
-
-                if (!int.TryParse(value, NumberStyles.Any, CultureInfo.InvariantCulture, out int result))
-                {
-                    return null;
-                }
-
-                return result;
-            }
-
             // Helper method that looks int the properties bag for the index of the passed in propertyname, and then returns that index from the args parameter.
-            static SyntaxNode? GetNode(ImmutableArray<IArgumentOperation> args, ImmutableDictionary<string, string?> properties, string propertyName, SyntaxGenerator generator, bool useOptionsMemberExpression, Compilation compilation, CancellationToken cancellationToken)
+            static SyntaxNode? GetNode(ImmutableDictionary<string, string?> properties, string propertyName, SyntaxGenerator generator, bool useOptionsMemberExpression)
             {
-                int? index = TryParseInt32(properties, propertyName);
-                if (index == null)
+                string? propertyValue = properties[propertyName];
+                if (propertyValue is null)
                 {
                     return null;
                 }
 
                 if (!useOptionsMemberExpression)
                 {
-                    return generator.LiteralExpression(args[index.Value].Value.ConstantValue.Value);
+                    return generator.LiteralExpression(propertyValue);
                 }
                 else
                 {
-                    RegexOptions options = (RegexOptions)(int)args[index.Value].Value.ConstantValue.Value;
-                    string optionsLiteral = Literal(options);
-                    return SyntaxFactory.ParseExpression(optionsLiteral).SyntaxTree.GetRoot(cancellationToken);
+                    string optionsLiteral = Literal(propertyValue);
+                    return SyntaxFactory.ParseExpression(optionsLiteral);
                 }
             }
 
-            static string Literal(RegexOptions options)
+            static string Literal(string regexOptionsKey)
```

Comment:
This should be renamed to something like `FormatRegexOptions`. It's not clear what it does by the name and parameter type.

Reply:
Probably `stringifiedRegexOptions` to make it clear this is a `RegexOptions` that we called `ToString` on?